core/src/main/scala/org/http4s/internal/package.scala

```diff
@@ -0,0 +1,95 @@
+/*
+ * Portions derived from https://github.com/functional-streams-for-scala/fs2/blob/v0.10.5/io/src/main/scala/fs2/io/JavaInputOutputStream.scala
+ * Copyright (c) 2013 Paul Chiusano, and respective contributors
+ */
+package org.http4s
+
+import cats.effect.{Async, Effect, Sync}
+import cats.implicits._
+import fs2.{Chunk, Sink, Stream}
+import java.io.{InputStream, OutputStream}
+import scala.concurrent.ExecutionContext
+
+package object internal {
+  private[http4s] def blocking[F[_], A](fa: F[A], blockingExecutionContext: ExecutionContext)(
+      implicit F: Async[F],
+      ec: ExecutionContext): F[A] =
+    for {
+      _ <- Async.shift[F](blockingExecutionContext)
+      att <- fa.attempt
+      _ <- Async.shift(ec)
+      fa0 <- F.fromEither(att)
+    } yield fa0
+
+  /** Like fs2.io.readInputStream, but does a double shift with [[blocking]]. */
```

Comment:
I think everything from here below disappears in fs2-1.0, because it does a double shift.  This is basically all a backport to fs2-0.10.